nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,631 @@
+        If standardize is True, the time-series are centered and normed:
+        their variance is put to 1 in the time dimension.
+
+    target_affine: 3x3 or 4x4 matrix, optional
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    target_shape: 3-tuple of integers, optional
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    mask_strategy: {'background' or 'epi'}, optional
+        The strategy used to compute the mask: use 'background' if your
+        images present a clear homogeneous background, and 'epi' if they
+        are raw EPI images. Depending on this value, the mask will be
+        computed from masking.compute_background_mask or
+        masking.compute_epi_mask. Default is 'background'.
+
+    memory: instance of joblib.Memory or string
+        Used to cache the masking process.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    n_jobs : int, optional. Default is -1.
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs'.
+
+    verbose : int, optional
+        Verbosity level. Default is False
+    """
+    # XXX problems when performing regression, it shold change to svr or
+    # ridge_regression
+    def __init__(self, estimator='svc', mask=None, cv=None, param_grid=None,
+                 screening_percentile=20, pos_label=None, scoring=None,
+                 smoothing_fwhm=None, standardize=True, target_affine=None,
+                 target_shape=None, mask_strategy='background',
+                 memory=Memory(cachedir=None), memory_level=0,
+                 n_jobs=1, verbose=False,
+                 ):
+        self.estimator = estimator
+        self.mask = mask
+        self.cv = cv
+        self.param_grid = param_grid
+        self.screening_percentile = screening_percentile
+        self.pos_label = pos_label
+        self.scoring = scoring
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.mask_strategy = mask_strategy
+        self.memory = memory
+        self.memory_level = memory_level
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+
+    def fit(self, niimgs, y, index=None):
+        """Fit the decoder
+
+        Parameters
+        ----------
+        niimgs: list of filenames or NiImages
+            Data on which the mask must be calculated. If this is a list,
+            the affine is considered the same for all.
+
+        y: 1D array-like
+           Target variable to predict. Must have exactly as many elements as
+           3D images in niimg.
```

Comment:
perhaps mention that continuous values for regression and discrete values for classification